addons/account/static/src/js/tax_group.js

```diff
@@ -0,0 +1,269 @@
+         * 
+         * @param {Array} idsArray
+         * @returns Parsed ids Array 
+         */
+        _getParsedIds: function (idsArray) {
+            var parsedId = []
+            idsArray.forEach(id => {
+                if (typeof(id) === "string" && !id.includes("virtual")) {
+                    id = parseInt(id.match(/\d+/)[0]);
+                }
+                parsedId.push(id);
+            });
+            return parsedId;
+        },
+
+        /**
+         * This method checks taht the document where the widget
+         * is located is of the "in_invoice" or "in_refund" type.
+         * This makes it possible to know if it is a purchase
+         * document.
+         * 
+         * @returns boolean (true if the invoice is a purchase document)
+         */
+        _isPurchaseDocument: function () {
+            return (this.recordData.type === "in_invoice" || this.recordData.type === 'in_refund');
+        },
+
+        /**
+         * This method is part of the widget life cycle and allows you to render 
+         * the widget.
+         * 
+         * @private
+         * @override 
+         */
+        _render: function () {
+            var self = this;
+            if (this.record.data.amount_by_group !== undefined) {
+                this.record.data.amount_by_group.forEach(amount_by_group => {
+                    if (amount_by_group[1] !== amount_by_group[8]) {
+                        amount_by_group[1] = amount_by_group[8];
+                        amount_by_group[3] = fieldUtils.format.monetary(amount_by_group[1], {}, { 'currency_id': self.record.data.currency_id.res_id });
+                    }
+                    amount_by_group[3] = amount_by_group[3].replace('&nbsp;', ' ');
+                });
+            }
+            this.$el.html($(QWeb.render('AccountTaxGroupTemplate', {
+                lines: self.record.data.amount_by_group,
+                isPurchaseDocument: self._isPurchaseDocument(),
+            })));
+        },
+
+        /**
+         * This method is called each time a field is changed.
+         * 
+         * @param {*} record 
+         * @param {*} event 
+         */
+        _reset: function (record, event) {
+            this._super.apply(this, arguments);
+            var self = this;
+            if (event.data.changes.invoice_line_ids) {
+                self.trigger_up('field_changed', {
+                    dataPointID: self.record.id,
+                    changes: { line_ids: record.data.line_ids },
+                });
+            }
+        },
+
+        /**
+         * This methode is responsible for modifying the value of the tax
+         * group by the new value entered by the user. It is also
+         * responsible for modifying taxes accordingly.
+         * 
+         * @param {event} ev 
+         */
+        _setTaxGroups: function (ev) {
+            var myInput;
+            if (ev.target.nodeName === 'INPUT') {
```

Comment:
this method should not have to work with dom element, you can change it in order for it to take 2 arguments, the tax_group_id and the value.